pkg/actor/validate_version.go

```diff
@@ -346,41 +421,67 @@ func IsContainerStatusImagePullBackoff(
 	pods, err := clientset.CoreV1().Pods(job.Namespace).List(ctx, metav1.ListOptions{
 		LabelSelector: labels.Set(job.Spec.Selector.MatchLabels).AsSelector().String(),
 	})
-	//TO DO: maybe we should check some k8s specific errors here
-	if err != nil {
-		return LogError("error getting pod in job", err, l)
+
+	if k8sErrors.IsNotFound(err) { // this is not an error
+		l.V(DEBUGLEVEL).Info("cannot find pods for vcheck job", "jobName", job.ObjectMeta.Name, "namespace", job.Namespace)
+		return err
+	} else if statusError, isStatus := err.(*k8sErrors.StatusError); isStatus { // this is an error
+		l.Error(statusError, fmt.Sprintf("status error getting pod %v", statusError.ErrStatus.Message))
+		return err
+	} else if err != nil {
+		l.V(int(zapcore.ErrorLevel)).Info("error finding pods for vcheck job", "jobName", job.ObjectMeta.Name, "namespace", job.Namespace)
+		return err
 	}
+
 	if len(pods.Items) == 0 {
-		return LogError("job pods are not running.", nil, l)
+		l.V(DEBUGLEVEL).Info("job pods are not running.")
+		return nil
 	}
 	pod := pods.Items[0]
 	if !kube.IsPodReady(&pod) && kube.IsImagePullBackOff(&pod, image) {
-		return LogError(fmt.Sprintf("Back-off pulling image %s", image), nil, l)
+		l.V(DEBUGLEVEL).Info(fmt.Sprintf("Back-off pulling image %s", image))
+		return nil
 	}
-	l.V(int(zapcore.DebugLevel)).Info("job pod is ready")
+	l.V(int(DEBUGLEVEL)).Info("job pod is ready")
```

Comment:
Changed, but I am not certain I like the flow then.

Reply:
Are there more than 2 states we care about? It seems like there are 3 states.
- we are in a backoff state
- we are not in a backoff state
- err because we can't find the job
It's more about what the caller gets back, if the pod is ready we should be good to proceed or at least make a different choice than if we're in a backoff state. Maybe the function is just misnamed and I'm making assumptions based on the name?

Reply:
With the backoff retries returning a tulip becomes quite messy.   A backoff retry only accepts a function that returns an error, but we could pass in a bool via a pointer.   We could return a custom error, but at the end of the day, we are returning the exact error up the stack.  I am not certain of the value we get with more complex code.